OlinkAnalyze/R/bridgeable.R

```diff
@@ -0,0 +1,293 @@
+                  # using 10% to 90% quantiles for checking range /IQR
+                  range_3k = quantile(.data[["NPX_3k"]][.data[["Count_3k"]] >
+                                                          10],
+                                      probs = 0.9) -
+                    quantile(.data[["NPX_3k"]][.data[["Count_3k"]] > 10],
+                             probs = 0.1),
+                  range_diff = abs(.data[["range_ht"]] - .data[["range_3k"]]),
+                  iqr_ht = IQR(.data[["NPX_ht"]]),
+                  iqr_3k = IQR(.data[["NPX_3k"]]),
+                  median_count_3k =
+                    median(.data[["Count_3k"]][.data[["Count_3k"]] > 10],
+                           na.rm = TRUE),
+                  # if count less than 10 in 1 sample that assay is dropped
+                  median_count_ht =
+                    median(.data[["Count_ht"]][.data[["Count_ht"]] > 10],
+                           na.rm = TRUE),
+                  # if count less than 10 in 1 sample that assay is dropped
+                  low_counts_flag = .data[["median_count_3k"]] < 150 |
+                    .data[["median_count_ht"]] < 150) |>
+    dplyr::ungroup() |>
+    dplyr::distinct()
+
+  # selecting only required columns
+  data_condensed <- data_mapped |>
+    dplyr::select(.data[["SampleID"]],
+                  .data[["Gene"]],
+                  .data[["UniProt"]],
+                  .data[["NPX_3k"]],
+                  .data[["NPX_ht"]],
+                  .data[["Count_ht"]],
+                  .data[["Count_3k"]],
+                  .data[["range_diff"]],
+                  .data[["low_counts_flag"]],
+                  .data[["ids"]]) |>
+    dplyr::ungroup() |>
+    dplyr::distinct() |>
+    r2_results()
+
+  not_bridgeable <- data_condensed |>
+    dplyr::filter(.data[["range_diff"]] >
+                    1, .data[["low_counts_flag"]] ==
+                    TRUE, .data[["R2_lm"]] < 0.8) |>
+    dplyr::pull(.data[["ids"]]) |>
+    unique() |>
+    as.data.frame() |>
+    dplyr::rename(ids = 1) |>
+    tidyr::extract(.data[["ids"]], into = c("OlinkID", "OlinkID_Explore384"),
+                   "(.*)_([^_]+)$", remove = TRUE)
+
+  # Adding type of normalization
+  ks <- ks_results(data = data_mapped) |>
+    tidyr::extract(.data[["ids"]], into = c("OlinkID", "OlinkID_Explore384"),
+                   "(.*)_([^_]+)$", remove = TRUE)
+
+  bridge_table <- e3k_eHT_mapping |>
+    dplyr::mutate(Bridgeable =
+                    dplyr::if_else(.data[["OlinkID_HT"]] %in%
+                                     not_bridgeable$OlinkID, "Not Bridgeable",
+                                   "Yes")) |>
+    dplyr::select(.data[["OlinkID_HT"]],
+                  .data[["OlinkID_E3072"]],
+                  .data[["Bridgeable"]]) |>
+    dplyr::distinct() |>
+    dplyr::inner_join(ks,
+                      by = c("OlinkID_HT" = "OlinkID",
+                             "OlinkID_E3072" = "OlinkID_Explore384"))
```

Comment:
I think not_bridgeable and ks could be part of the same pipe, and we would not need all these intermediate datasets.